src/workerd/io/io-context.h

```diff
@@ -21,6 +21,9 @@
 #include <kj/function.h>
 #include <capnp/dynamic.h>
 #include <workerd/util/weak-refs.h>
+#include <workerd/io/limit-enforcer.h>
+#include <workerd/io/io-channels.h>
+#include <workerd/util/tracing.h>
```

Comment:
I don't think we can include this header in headers which might be included by embedding projects.

Specifically, having two PERFETTO_DEFINE_CATEGORIES_IN_NAMESPACE macros in the same translation will cause the following error: `error: target of using declaration conflicts with declaration already in scope`.

Reply:
I *think* this might actually have to do with the aliasing that we're doing with the internal repo. That is, we alias `workerd::*` to the internal namespace. That could be conflicting with this if the two sets of categories end up in the same namespace?

Reply:
ok, nevermind, I see what you're saying here. Not sure how we can handle this since we absolutely want to be able to instrument the inline functions in files like `io-context.h` and we need the categories registered for that to work. What I'm wondering is if we need to have the internal repo declare the `workerd` category in it's own storage, then only register these specifically in the `workerd` files if it's a standalone build.

Reply:
Oh! ok, I got it. The issue is not the `PERFETTO_DEFINE_CATEGORIES_IN_NAMESPACE`, it's the `PERFETTO_USE_CATEGORIES_FROM_NAMESPACE` that is the issue. We can have as many `PERFETTO_DEFINE_CATEGORIES_IN_NAMESPACE` that we want but we can only have one `..._USE_CATEGORIES_...` per translation unit.